gdnative-core/src/export/property/hint.rs

```diff
@@ -116,30 +116,36 @@ where
 /// ```
 #[derive(Clone, Eq, PartialEq, Debug, Default)]
 pub struct EnumHint {
-    values: Vec<String>,
+    values: Vec<(String, Option<i64>)>,
 }
 
 impl EnumHint {
     #[inline]
     pub fn new(values: Vec<String>) -> Self {
+        let values = values.into_iter().map(|v| (v, None)).collect();
+        EnumHint { values }
+    }
+
+    #[inline]
+    pub fn with_numbers(values: Vec<(String, i64)>) -> Self {
+        let values = values
+            .into_iter()
+            .map(|(key, val)| (key, Some(val)))
+            .collect();
         EnumHint { values }
     }
 
     /// Formats the hint as a Godot hint string.
     fn to_godot_hint_string(&self) -> GodotString {
-        let mut s = String::new();
-
-        let mut iter = self.values.iter();
-
-        if let Some(first) = iter.next() {
-            write!(s, "{}", first).unwrap();
-        }
-
-        for rest in iter {
-            write!(s, ",{}", rest).unwrap();
-        }
-
-        s.into()
+        self.values
+            .iter()
+            .map(|(key, val)| match val {
+                Some(val) => format!("{}:{}", key, val),
+                None => key.clone(),
+            })
+            .collect::<Vec<_>>()
+            .join(",")
+            .into()
```

Comment:
The original code is iterative for a reason. While `format!`, `collect` and `join` are indeed convenient and pretty, they perform many allocations under the hood. It's often said that premature optimization is the root of all evil, but that does not mean we need to needlessly slow things down just because.

Reply:
Done